src/cripts/README.md

```diff
@@ -0,0 +1,453 @@
+
+## 3.3. Processing and modifying headers
+
+A big part of all Cripts is to read and modify various headers. ATS as it works,
+has four distinct headers, which translates into the following four environments:
+
+| Header name      | Description                                        |
+| ---------------- | -------------------------------------------------- |
+| Client::Request  | The clients request header                         |
+| Client::Response | The response header that is sent to the client     |
+| Server::Request  | The request header being sent to the origin server |
+| Server::Response | The response header being received from origin     |
+
+Accessing these headers is easy:
+```
+    borrow client_req  = Client::Request::get();
+    borrow client_resp = Client::Response::get();
+    borrow server_req  = Server::Request::get();
+    borrow server_resp = Server::Response::get();
+```
+
+Note that not all of these are available in every callback; For example, the client
+response header is not available to read or write until we are in the hook for
+`do_send_response()`.
+
+The response headers has a couple of additional features specifically for
+responses:
+```
+borrow client_resp = Client::Response::get();
+
+if (resp.status = 220) {
+    resp.status = 200;
+}
+
+// TBD more stuff here ?
+```
+
+Similarly, the request headers has a few unique traits as well:
+
+```
+borrow client_req = Client::Request::get();
+
+if (client_req.method == "GET") {
+    ...
+}
+
+// TBD more stuff here ?
+```
+
+## 3.4. Processing and modifying URLs
+
+Similarly to headers, URLs are important in Cripts, in all its various forms.
+Currently, Cript supports the following URLs:
+
+| URL name         | Description                                                    |
+| ---------------- | -------------------------------------------------------------- |
+| Client::Pristine | The pristine client request URL, which is immutable            |
+| Client::URL      | The clients request URL, which can be modified                 |
+| Cache::URL       | This is a special URL, used internally of ATS as the cache key |
+
+Getting these URLs follows a similar getter pattern to the headers:
+```
+borrow pristine = Pristine::URL::get();
+borrow client   = Client::URL::get();
+borrow cache    = Cache::URL::get();
+```
+
+Within a URL object, you can access all its various components via names following
+standard URL naming.
+
+| Identifier | Description                             |
+| ---------- | --------------------------------------- |
+| host       | The URL host                            |
+| port       | The URL port number                     |
+| path       | The URL path                            |
+| query      | The URL query parameters (all of them!) |
+
+Using this is easy (of course):
+```
+auto url = Client::URL;
+
+if (url.host == "www.example.com") {
+    ...
+}
+```
+
+The `path` components does support indexes, e.g. the first URL path
```

Comment:
I suggest `indexing` rather than `indexes`. If we stick with the plural of `index`, `indices` is more common in technical docs.